substrate/frame/identity/src/types.rs

```diff
@@ -319,6 +324,29 @@ pub struct RegistrarInfo<
 	pub fields: IdField,
 }
 
+/// The number of usernames that an authority may allocate.
+type Allocation = u32;
+/// The maximum allowed length of a suffix.
+const SUFFIX_MAX_LENGTH: u32 = 7;
+/// A byte vec used to represent a username.
+pub(crate) type Suffix = BoundedVec<u8, ConstU32<SUFFIX_MAX_LENGTH>>;
+
+/// Properties of a username authority.
+#[derive(Clone, Encode, Decode, MaxEncodedLen, TypeInfo, PartialEq, Debug)]
+pub struct AuthorityProperties {
+	/// The suffix added to usernames granted by this authority. Will be appended to usernames; for
+	/// example, a suffix of `wallet` will result in `.wallet` being appended to a user's selected
+	/// name.
+	pub suffix: Suffix,
+	/// The number of usernames remaining that this authority can grant.
+	pub allocation: Allocation,
+}
+
+/// The maximum allowed length of a username, _including_ the suffix.
+pub(crate) const USERNAME_MAX_LENGTH: u32 = 32;
```

Comment:
```suggestion
pub const USERNAME_MAX_LENGTH: u32 = 32;
```
I think these should be pub, otherwise it will be difficult for externals to even discover these values.